File: day01/main.py
def get_input():
    with open("input.txt", "r") as f:
        lines = f.read().splitlines()
        return lines

lines = get_input()

totals = []
total = 0
for line in lines:
    if line == '':
        totals.append(total)
        total = 0
    else:
        total += int(line)
print(max(totals))
print(sum(sorted(totals)[-3:]))

File: day01/main.js
const fs = require('fs');

// Read from input.txt
const getInput = () => fs.readFileSync(`${__dirname}/input.txt`).toString().split('\n');

const sum = (arr) => arr.reduce((memo, n) => (memo + n), 0)

// Solve the puzzle and output the solution
const main = () => {
    const lines = getInput();

    let totals = [];
    let total = 0;
    for (const line of lines) {
        console.log(line)
        if (line === "") {
            totals.push(total)
            total = 0;
        } else {
            total += parseInt(line)
        }
    }
    totals.sort()
    console.log(Math.max(...totals))
    console.log(sum(totals.slice(-3)))
}

main();
